diff --git a/shared/device.mk b/shared/device.mk
index 9cb75afa7..1f3e6d9f3 100644
--- a/shared/device.mk
+++ b/shared/device.mk
@@ -360,7 +360,7 @@ PRODUCT_PACKAGES += \
 # Weaver aidl HAL
 #
 PRODUCT_PACKAGES += \
-    android.hardware.weaver-service.example
+    android.hardware.weaver-service.default
 
 #
 # IR aidl HAL
diff --git a/shared/sepolicy/vendor/file_contexts b/shared/sepolicy/vendor/file_contexts
index 8e536e30d..c7b57ef1f 100644
--- a/shared/sepolicy/vendor/file_contexts
+++ b/shared/sepolicy/vendor/file_contexts
@@ -107,7 +107,7 @@
 /vendor/bin/hw/android\.hardware\.gatekeeper-service.remote  u:object_r:hal_gatekeeper_remote_exec:s0
 /vendor/bin/hw/android\.hardware\.confirmationui-service.cuttlefish  u:object_r:hal_confirmationui_cuttlefish_exec:s0
 /vendor/bin/hw/android\.hardware\.oemlock-service.example u:object_r:hal_oemlock_default_exec:s0
-/vendor/bin/hw/android\.hardware\.weaver-service.example u:object_r:hal_weaver_default_exec:s0
+/vendor/bin/hw/android\.hardware\.weaver-service.default u:object_r:hal_weaver_default_exec:s0
 /vendor/bin/hw/android\.hardware\.authsecret@1\.0-service  u:object_r:hal_authsecret_default_exec:s0
 /vendor/bin/hw/android\.hardware\.authsecret-service.example u:object_r:hal_authsecret_default_exec:s0
 /vendor/bin/hw/android\.hardware\.rebootescrow-service\.default  u:object_r:hal_rebootescrow_default_exec:s0
