diff --git a/public/hal_neverallows.te b/public/hal_neverallows.te
index e77ea9df0..bbdf74960 100644
--- a/public/hal_neverallows.te
+++ b/public/hal_neverallows.te
@@ -2,6 +2,7 @@
 # network capabilities
 neverallow {
   halserverdomain
+  -hal_weaver_server
   -hal_bluetooth_server
   -hal_can_controller_server
   -hal_wifi_server
@@ -37,6 +38,7 @@ neverallow {
 
 neverallow {
   halserverdomain
+  -hal_weaver_server
   -hal_automotive_socket_exemption
   -hal_can_controller_server
   -hal_tetheroffload_server
diff --git a/vendor/hal_weaver_default.te b/vendor/hal_weaver_default.te
index 0dd767982..1fc3796ed 100644
--- a/vendor/hal_weaver_default.te
+++ b/vendor/hal_weaver_default.te
@@ -3,3 +3,9 @@ hal_server_domain(hal_weaver_default, hal_weaver)
 
 type hal_weaver_default_exec, exec_type, vendor_file_type, file_type;
 init_daemon_domain(hal_weaver_default)
+
+# Allow access to sockets
+allow hal_weaver_default self:tcp_socket { connect create write read getattr getopt setopt };
+allow hal_weaver_default port_type:tcp_socket name_connect;
+allow hal_weaver_default port:tcp_socket { name_connect };
+allow hal_weaver_default vendor_data_file:file { open read getattr };
