diff --git a/vendor/file_contexts b/vendor/file_contexts
index 0cfb7cf39..74aef4bd0 100644
--- a/vendor/file_contexts
+++ b/vendor/file_contexts
@@ -82,6 +82,7 @@
 /(vendor|system/vendor)/bin/hw/android\.hardware\.sensors@[0-9]\.[0-9]-service(\.multihal)?  u:object_r:hal_sensors_default_exec:s0
 /(vendor|system/vendor)/bin/hw/android\.hardware\.sensors-service\.example  u:object_r:hal_sensors_default_exec:s0
 /(vendor|system/vendor)/bin/hw/android\.hardware\.secure_element@1\.0-service u:object_r:hal_secure_element_default_exec:s0
+/(vendor|system/vendor)/bin/hw/android\.hardware\.weaver@1\.0-service u:object_r:hal_weaver_default_exec:s0
 /(vendor|system/vendor)/bin/hw/android\.hardware\.security\.dice-service\.non-secure-software   u:object_r:hal_dice_default_exec:s0
 /(vendor|system/vendor)/bin/hw/android\.hardware\.security\.keymint-service   u:object_r:hal_keymint_default_exec:s0
 /(vendor|system/vendor)/bin/hw/rild                                           u:object_r:rild_exec:s0
diff --git a/vendor/hal_weaver_default.te b/vendor/hal_weaver_default.te
index 0dd767982..9a8bbc040 100644
--- a/vendor/hal_weaver_default.te
+++ b/vendor/hal_weaver_default.te
@@ -3,3 +3,5 @@ hal_server_domain(hal_weaver_default, hal_weaver)
 
 type hal_weaver_default_exec, exec_type, vendor_file_type, file_type;
 init_daemon_domain(hal_weaver_default)
+
+allow hal_weaver_default secure_element_service:service_manager find
